src/frequenz/sdk/timeseries/_grid_frequency.py

```diff
@@ -49,32 +49,30 @@ def __init__(
         self,
         data_sourcing_request_sender: Sender[ComponentMetricRequest],
         channel_registry: ChannelRegistry,
-        component: Component,
+        source: Component,
```

Comment:
So basically `source: Component | None` and do the find if it is `None`, or even just removing the `source` too, not sure if it is worth keeping. Any opinions @shsms @matthias-wende-frequenz ?

Reply:
I like the idea of removing the param and identifying the right component within the class.  The only way it will get configurable is if we decide to do specify a component ID through the UI, which could happen at runtime.  So it will have to be implemented differently anyway.